### PR TITLE
Add subscriber count to iframe event handler events

### PR DESCRIPTION
**Pull request by adammw**
:v:

/cc @zendesk/vegemite

### Description
Add subscriber count to iframe event handler events.
This allows the framework to know how many subscribers are listening inside the iframe, and when it's safe to bind/unbind.

### References
* JIRA: 

### Risks
* low: data attribute of event is unused

### CODE REVIEW

**Comment by danielbreves** [comment]
👍

**Comment by svizzari** [comment]
👍

**Comment by sandlerr** [comment]
@zendesk/vegemite I just pushed one more commit, can we have one more thumb please?

**Comment by svizzari** [comment]
>  I just pushed one more commit, can we have one more thumb please?

oh, yea, that'd be bad if we started splicing elements from the end of the array when start index was negative

👍

File: lib/client.js [hunk 1]
```diff
@@ -248,7 +248,9 @@ Client.prototype = {
     if (this._parent) { throw new Error('Not Implemented'); }
     if (!this._messageHandlers[name]) { return false; }
     var index = this._messageHandlers[name].indexOf(handler);
-    this._messageHandlers[name].splice(index, 1);
+    if (this.has(name, handler)) {
```
Comment (adammw) [inline lib/client.js]: wouldn't `index !== -1` be equivilant? #codegolf

**Pull request merged**